packages/xforms-engine/src/lib/reactivity/text/createTextRange.ts

```diff
@@ -1,128 +1,72 @@
 import { createMemo } from 'solid-js';
-import type { TextMediaSource, TextRole } from '../../../client/TextRange.ts';
+import type { TextRole } from '../../../client/TextRange.ts';
 import type { EvaluationContext } from '../../../instance/internal-api/EvaluationContext.ts';
 import { TextChunk } from '../../../instance/text/TextChunk.ts';
-import { TextRange } from '../../../instance/text/TextRange.ts';
-import type {
-	EngineXPathAttribute,
-	EngineXPathNode,
-} from '../../../integration/xpath/adapter/kind.ts';
-import type { TextChunkExpression } from '../../../parse/expression/TextChunkExpression.ts';
+import { TextRange, type MediaSources } from '../../../instance/text/TextRange.ts';
+import { isEngineXPathElement } from '../../../integration/xpath/adapter/kind.ts';
+import { StaticElement } from '../../../integration/xpath/static-dom/StaticElement.ts';
+import { type TextChunkExpression } from '../../../parse/expression/TextChunkExpression.ts';
 import type { TextRangeDefinition } from '../../../parse/text/abstract/TextRangeDefinition.ts';
 import { createComputedExpression } from '../createComputedExpression.ts';
 
-interface TextContent {
+interface ChunksAndMedia {
 	chunks: readonly TextChunk[];
-	mediaSource: TextMediaSource | undefined;
+	mediaSources: MediaSources;
 }
 
-const isElementNode = (
-	node: EngineXPathNode | string
-): node is EngineXPathNode & {
-	attributes: EngineXPathAttribute[];
-	children: EngineXPathNode[];
-	value?: string;
-} => {
-	return typeof node !== 'string' && 'children' in node && 'value' in node && 'attributes' in node;
-};
-
-const isTextNode = (
-	node: EngineXPathNode | string
-): node is EngineXPathNode & {
-	children: EngineXPathNode[];
-	value?: string;
-} => {
-	return (
-		typeof node !== 'string' && 'children' in node && 'value' in node && !('attributes' in node)
-	);
-};
-
-const isFormAttribute = (attribute: EngineXPathAttribute) => {
-	return attribute?.qualifiedName?.localName === 'form';
-};
-
-const isDefaultValue = (item: EngineXPathNode | string) => {
-	return (isElementNode(item) && !item.attributes?.length) || isTextNode(item);
-};
-
-const getMediaSource = (item: EngineXPathNode): TextMediaSource | undefined => {
-	if (isDefaultValue(item) || !isElementNode(item)) {
-		return;
-	}
-
-	const value = item.value ?? '';
-	const mediaType = item.attributes.find(isFormAttribute)?.value;
-
-	return {
-		image: mediaType === 'image' ? value : '',
-		video: mediaType === 'video' ? value : '',
-		audio: mediaType === 'audio' ? value : '',
-	};
-};
-
-/**
- * The function temporarily supports a <value> node with multiple nested nodes.
- * TODO: Build support for <output> nodes.
- *  A child might be a node that needs XPath to compute its value. For that, the engine
- *  should create a {@link: TextElementDefinition}, so that createTextChunks function can
- *  request the computed value to XPath and create the TextChunk.
- */
-const processValueNodeChildren = (item: EngineXPathNode) => {
```

Comment:
If I recall correctly, this was introduced to re-establish passable behavior for itext default forms with outputs in them. I don't think it's necessary for this PR or if we do want to do it here we should introduce a test related to it.